diff --git a/caliper/src/main/java/com/google/caliper/runner/Instrument.java b/caliper/src/main/java/com/google/caliper/runner/Instrument.java
index 0f32fa9..4cd26af 100644
--- a/caliper/src/main/java/com/google/caliper/runner/Instrument.java
+++ b/caliper/src/main/java/com/google/caliper/runner/Instrument.java
@@ -153,7 +153,8 @@ public abstract class Instrument {
       // do compilation serially
       "-Xbatch",
       // make sure compilation doesn't run in parallel with itself
-      "-XX:CICompilerCount=1",
+      // See https://bugs.openjdk.java.net/browse/JDK-8130858
+      "-XX:CICompilerCount=2",
       // ensure the parallel garbage collector
       "-XX:+UseParallelGC",
       // generate classes or don't, but do it immediately
