diff --git a/pom.xml b/pom.xml
index 76d7547..50fef9a 100644
--- a/pom.xml
+++ b/pom.xml
@@ -40,8 +40,8 @@
 
 	<properties>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-		<maven.compiler.target>11</maven.compiler.target>
-		<maven.compiler.source>11</maven.compiler.source>
+		<maven.compiler.target>8</maven.compiler.target>
+		<maven.compiler.source>8</maven.compiler.source>
 		<xcodeScheme>Debug</xcodeScheme>
 	</properties>
 
diff --git a/src/main/java/ca/weblite/objc/Proxy.java b/src/main/java/ca/weblite/objc/Proxy.java
index 0654519..7aba1f5 100644
--- a/src/main/java/ca/weblite/objc/Proxy.java
+++ b/src/main/java/ca/weblite/objc/Proxy.java
@@ -501,7 +501,6 @@ public class Proxy implements Peerable {
      */
     @Override
     public String toString(){
-        System.out.println("The peer is "+getPeer());
         if ( getPeer() == null ){
             return "null";
         }
