diff --git a/third-party/java/aosp/src/com/android/manifmerger/OtherOperationType.java b/third-party/java/aosp/src/com/android/manifmerger/OtherOperationType.java
index d0ca70b544..4e21ee79ae 100644
--- a/third-party/java/aosp/src/com/android/manifmerger/OtherOperationType.java
+++ b/third-party/java/aosp/src/com/android/manifmerger/OtherOperationType.java
@@ -26,5 +26,5 @@ public enum OtherOperationType {
     ignore,
 
     // used to direct lint
-    targetAPI
+    targetapi // deliberately lowercase because we do lowercase before valueOf call
 }
diff --git a/third-party/java/aosp/src/com/android/manifmerger/XmlElement.java b/third-party/java/aosp/src/com/android/manifmerger/XmlElement.java
index fe9a0db424..f9ef270794 100644
--- a/third-party/java/aosp/src/com/android/manifmerger/XmlElement.java
+++ b/third-party/java/aosp/src/com/android/manifmerger/XmlElement.java
@@ -33,6 +33,7 @@ import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.Lists;
 
+import java.util.Locale;
 import org.w3c.dom.Attr;
 import org.w3c.dom.Element;
 import org.w3c.dom.NamedNodeMap;
@@ -126,7 +127,7 @@ public class XmlElement extends OrphanXmlElement {
                     } catch (IllegalArgumentException e) {
                         try {
                             // is this another tool's operation type that we do not care about.
-                            OtherOperationType.valueOf(instruction);
+                            OtherOperationType.valueOf(instruction.toLowerCase(Locale.ROOT));
                             break;
                         } catch (IllegalArgumentException e1) {
 
