diff --git a/third-party/java/aosp/src/com/android/common/sdklib/internal/build/SignedJarBuilder.java b/third-party/java/aosp/src/com/android/common/sdklib/internal/build/SignedJarBuilder.java
index 8209c4d9c7..8023afb480 100644
--- a/third-party/java/aosp/src/com/android/common/sdklib/internal/build/SignedJarBuilder.java
+++ b/third-party/java/aosp/src/com/android/common/sdklib/internal/build/SignedJarBuilder.java
@@ -182,13 +182,19 @@ public class SignedJarBuilder {
         String name = entry.getName();
 
         // do not take directories or anything inside a potential META-INF folder.
-        if (entry.isDirectory() || name.startsWith("META-INF/")) {
+        if (entry.isDirectory()) {
+          continue;
+        }
+     
+        if (name.startsWith("META-INF/") && !name.startsWith("META-INF/services/")) {
           continue;
         }
 
         // if we have a filter, we check the entry against it
-        if (filter != null && filter.checkEntry(name) == false) {
-          continue;
+        if (filter != null && filter.checkEntry(name) == false) { 
+          if (!name.startsWith("META-INF/services/")) {
+            continue;
+          }
         }
 
         JarEntry newEntry;
